mobile/lib/src/domain/sleep_sequence/sleep_sequence_metrics.dart

```diff
@@ -0,0 +1,25 @@
+class SleepSequenceMetrics {
+  Duration effectiveSleepTime;
+  double sleepEfficiency;
+  int sleepLatency;
+  Duration waso;
+  int awakenings;
+  int remLatency;
+  int shifts;
+
+  SleepSequenceMetrics({
+    this.effectiveSleepTime,
+    this.sleepEfficiency,
+    this.sleepLatency,
+    this.waso,
+    this.awakenings,
+    this.remLatency,
+    this.shifts,
+  })  : assert(!effectiveSleepTime.isNegative),
```

Comment:
Plutôt que de faire des assert pourquoi ne pas throw une erreur custom? :smile:

Reply:
Ou bien ajouter une string au assert https://dart.dev/guides/language/language-tour#assert

Reply:
La seule chose avec les asserts est qu'ils ne sont pas évalué en production. Donc, si c'est prévu qu'il y ait une exception en production et que c'est dans le fonctionnement du programme, il faut faire attention :)

Reply:
Design choice imo. On pensait ajouter l'option de re-analyze quand on a une nouvelle version du modèle, so l'usager pourrait lui-même réanalysé si par erreur le serveur retourne quelque chose de nulle tant que ça brise pas l'application. On peut le laisser en suspend (mettre assert null for now) et une fois qu'on teste la partie dashboard on peut voir si ça brise, ou on fait un choix live.